### DISCUSSION TITLE
Are dates not parsed when retrieved?

### DISCUSSION THREAD

**Discussion by sneko** [timgit/pg-boss#417] [Q&A]
Hi @timgit ,

I'm sending an event with a date into its data, but when retrieved I get them stringified.

I do understand it's stringified into the database due to JSON not having a "Date" type. But did I miss something so your library parses the event data dates since "Date" is kind of a primitive type?

If not, what do you suggest (I'm using `zod` to validate the data so I plan to go with `z.coerce.date()` instead of `z.date()` to force conversion)? Maybe a documentation warning is worth it?

Thank you,

**Comment by timgit** [comment]
Yes, all data is stringified to JSON to save into postgres.  I don't work with zod very much, but it sounds like z.coerce.date() or z.string().datetime() might be fine.
